test/e2e/e2e.go

```diff
@@ -742,6 +748,504 @@ var _ = Describe("validation", func() {
+				}, timeout, interval).Should(Succeed())
+
+				// Checking ConfigMap is created
+				By("By checking ConfigMap is created match the expected configuration")
+				Eventually(func() (string, error) {
+					configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+					if err != nil {
+						return "", err
+					}
+					return configmap.Data[consts.MetalLBConfigMapName], err
+				}, timeout, interval).Should(MatchYAML(`address-pools:
+- name: addresspool1
+  protocol: bgp
+  addresses:
+
+  - 10.1.1.1
+  - 10.1.1.100
+
+- name: addresspool2
+  protocol: bgp
+  addresses:
+
+  - 20.2.2.1
+  - 20.2.2.100
+peers:
+- my-asn: 60500
+  peer-address: 11.0.0.1
+  peer-asn: 60501
+  router-id: 11.11.11.11
+`))
+			})
+			By("By creating BGP peer2", func() {
+				peer := &metallbv1alpha1.BGPPeer{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "peer2",
+						Namespace: OperatorNameSpace,
+					},
+					Spec: metallbv1alpha1.BGPPeerSpec{
+						Address:  "12.0.0.1",
+						ASN:      60601,
+						MyASN:    60600,
+						RouterID: "12.12.12.12",
+					},
+				}
+				Expect(testclient.Client.Create(context.Background(), peer)).Should(Succeed())
+
+				key := types.NamespacedName{
+					Name:      "peer2",
+					Namespace: OperatorNameSpace,
+				}
+				// Create BGP Peer resource
+				By("By checking BGP peer resource is created")
+				Eventually(func() error {
+					err := testclient.Client.Get(context.Background(), key, peer)
+					return err
+				}, timeout, interval).Should(Succeed())
+
+				// Checking ConfigMap is created
+				By("By checking ConfigMap is created match the expected configuration")
+				Eventually(func() (string, error) {
+					configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+					if err != nil {
+						return "", err
+					}
+					return configmap.Data[consts.MetalLBConfigMapName], err
+				}, timeout, interval).Should(MatchYAML(`address-pools:
+- name: addresspool1
+  protocol: bgp
+  addresses:
+
+  - 10.1.1.1
+  - 10.1.1.100
+
+- name: addresspool2
+  protocol: bgp
+  addresses:
+
+  - 20.2.2.1
+  - 20.2.2.100
+peers:
+- my-asn: 60500
+  peer-address: 11.0.0.1
+  peer-asn: 60501
+  router-id: 11.11.11.11
+- my-asn: 60600
+  peer-address: 12.0.0.1
+  peer-asn: 60601
+  router-id: 12.12.12.12
+
+`))
+			})
+		})
+		It("should have deleted addresspools", func() {
+			By("Deleting first addresspool object ", func() {
+				addresspool := &metallbv1alpha1.AddressPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "addresspool1",
+						Namespace: OperatorNameSpace,
+					},
+					Spec: metallbv1alpha1.AddressPoolSpec{
+						Protocol: "bgp",
+						Addresses: []string{
+							"10.1.1.1",
+							"10.1.1.100",
+						},
+					},
+				}
+
+				Eventually(func() bool {
+					err := testclient.Client.Delete(context.Background(), addresspool)
```

Comment:
might be better delete the resource once (before the eventually) and then inside the eventually use Get to check if the error is NotFound.